src/samarium/shell.py

```diff
@@ -9,28 +16,129 @@
 INDENT = dahlia("&3  > ")
 
 
-def read_statement() -> str:
-    statement = ""
-    prompt = IN
-    while True:
-        statement += input(prompt)
-        if statement:
-            statement += " ;"
-            error, _ = match_brackets(tokenize(statement))
-            if not error:
-                return statement
-            prompt = INDENT
-            statement = statement[:-1]
-
-
-def run_shell(debug: bool):
-    dprint(f"&3Samarium {__version__}")
-    MAIN = Registry(globals())
-    while True:
-        try:
-            run(read_statement(), MAIN, debug, load_template=False, quit_on_error=False)
-            MAIN.output *= 0
-        except KeyboardInterrupt:
-            print()
-        except EOFError:
-            break
+class FakeIO:
+    def __init__(self) -> None:
+        self.output: list[str] = []
+
+    def write(self, __s: str) -> int:
+        self.output.append(__s)
+        return len(__s)
```

Comment:
Why `__s` and not just `s`?

Reply:
Well originally this was subclassing StringIO and when I used intelisense to autocomplete the method it chose that variable name.